### PR TITLE
docs(headless): add React samples for headless search parameter manager

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-415

### CODE REVIEW

File: packages/samples/headless-react/src/App.tsx [hunk 1]
```diff
@@ -10,7 +12,16 @@ import {ResultList} from './components/result-list/result-list.class';
 import {ResultList as ResultListFn} from './components/result-list/result-list.fn';
 import {Pager} from './components/pager/pager.class';
 import {Pager as PagerFn} from './components/pager/pager.fn';
+import {ResultsPerPage} from './components/results-per-page/results-per-page.class';
```
Comment (btaillon) [inline packages/samples/headless-react/src/App.tsx:15]: reordered

**Review by ThibodeauJF — APPROVED** [review]

File: packages/samples/headless-react/src/App.tsx [hunk 2]
```diff
@@ -64,6 +76,14 @@ const resultsPerPage = buildResultsPerPage(engine, {
 const pager = buildPager(engine, {options: {numberOfPages: 6}});
 
 function App() {
+  useEffect(
+    () =>
+      searchParameterManager.subscribe(() =>
+        writeSearchParametersToURI(searchParameterManager.state.parameters)
+      ),
+    []
+  );
```
Comment (samisayegh) [inline packages/samples/headless-react/src/App.tsx:85]: Could we place the subscription in the same file as the serializer? The way these samples will be displayed is by associating a controller with one or more files containing samples. If one of the files is App.tsx though, then readers will see a large file and not know where to look?

If we could show the interaction in single file, or two focused files, the sample will be much clearer.

**Review by olamothe — APPROVED** [review]
Could have been cool to also have an example with popular routing library 🤔 ...

File: packages/samples/headless-react/src/components/search-parameter-manager/search-parameter-manager.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    initialState: {parameters: readSearchParametersFromURI()},
+  });
+
+  return {
+    autoUpdateURI: () =>
+      searchParameterManager.subscribe(() =>
+        writeSearchParametersToURI(searchParameterManager.state.parameters)
+      ),
+  };
+}
```
Comment (samisayegh) [inline packages/samples/headless-react/src/components/search-parameter-manager/search-parameter-manager.ts:30]: Unlike production code, I feel sample code is much easier to read if there was only one function. I would suggest:

```
export function bindSearchParametersToURI(engine: Engine) {
    const hash = window.location.hash.slice(1);
    const parameters = deserialize(decodeURIComponent(hash))

    const manager = buildSearchParameterManager(engine, {initialState: {parameters}};

   ...
}
```

**Review by samisayegh — APPROVED** [review]

**Comment by samisayegh** [comment]
> Could have been cool to also have an example with popular routing library

I think this is specialized use-case to have as the reference sample. I think we should have a sample for it but place it in a dedicated section under e.g. `commerce/seo-friendly-urls`

**Pull request merged**